components/script/dom/dommatrix.rs

```diff
@@ -0,0 +1,343 @@
+        fn M43(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().M43()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-m43
+        fn SetM43(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m43(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-m44
+        fn M44(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().M44()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-m44
+        fn SetM44(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m44(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-a
+        fn A(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().A()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-a
+        fn SetA(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m11(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-b
+        fn B(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().B()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-b
+        fn SetB(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m12(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-c
+        fn C(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().C()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-c
+        fn SetC(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m21(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-d
+        fn D(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().D()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-d
+        fn SetD(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m22(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-e
+        fn E(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().E()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-e
+        fn SetE(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m41(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-f
+        fn F(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().F()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-f
+        fn SetF(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m42(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrix-multiplyself
+        fn MultiplySelf(&self, other:&DOMMatrixInit) -> Root<DOMMatrix> {
```

Comment:
@nox The various methods on `DOMMatrixReadOnly` that are actually really there to be used by `DOMMatrix` all now have a return value. However, this is a `DOMMatrixReadOnly`. I can't see a clean way to make this value be used by the `DOMMatrix` methods. Currently the implementation is technically correct because these DOM methods return the current object, but the implementations (e.g. [multiply_self](https://github.com/servo/servo/pull/12202/files#diff-5c52ec6cb0bad6e8c9e0ee45c26cf378R155)) are a bit misleading because they return a value that is never used.